test/libsolidity/syntaxTests/constants/cyclic_dependency_file_constants.sol

```diff
@@ -0,0 +1,13 @@
+==== Source: A.sol ====
+import "B.sol";
+
+uint256 constant A = B + 1;
+
+==== Source: B.sol ====
+import "A.sol";
+
+uint256 constant B = A + 1;
```

Comment:
These cyclic cases are fine but we also need some that will actuallyl compile and where the visiting order could have an impact. For example something like this:

```solidity
==== Source: A.sol ====
import "B.sol";

library L {
    uint constant X = 1;
    uint constant Y = K.Y;
}
==== Source: B.sol ====
import "A.sol";

library K {
    uint constant X = L.X;
    uint constant Y = 2;
}
```

or
```solidity
==== Source: A.sol ====
import "B.sol" as B;

uint constant X = 1;
uint constant Y = B.Y;
==== Source: B.sol ====
import "A.sol" as A;

uint constant X = A.X;
uint constant Y = 2;
```

Reply:
Added as test case.

Reply:
Can you add the one with libraries too?

Reply:
Ah I missed it. Done.